### PR TITLE
python3Packages.rsskey: init at 0.2.0

### PR DESCRIPTION
**Pull request by McSinyx**
###### Description of changes

Add rsskey and its dependency markdownify

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (`python -m rsskey`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 185118` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.markdownify</li>
    <li>python310Packages.rsskey</li>
    <li>python39Packages.markdownify</li>
    <li>python39Packages.rsskey</li>
  </ul>
</details>

File: pkgs/development/python-modules/markdownify/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  version = "0.11.2";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-1hP5WyZJ2Lg8TEyy5sJM6ox99OB9QY+S+iWtDduwReA=";
+  };
+
+  nativeBuildInputs = [ flake8 ];
+  propagatedBuildInputs = [ beautifulsoup4 six ];
+  checkInputs = [ pytestCheckHook pytest ];
```
[the remark this answers is not in the document]
Comment (McSinyx) [reply pkgs/development/python-modules/markdownify/default.nix]: You're right, it is.

File: pkgs/development/python-modules/rsskey/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{ lib, buildPythonPackage, fetchPypi
+, feedparser, httpx, loca, markdownify, trio }:
```
[the remark this answers is not in the document]
Comment (McSinyx) [reply pkgs/development/python-modules/rsskey/default.nix]: Not my cup of tea but the dep list is short enough here so :shrug:

File: pkgs/development/python-modules/markdownify/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+buildPythonPackage rec {
+  pname = "markdownify";
+  version = "0.11.2";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-1hP5WyZJ2Lg8TEyy5sJM6ox99OB9QY+S+iWtDduwReA=";
+  };
+
+  nativeBuildInputs = [ flake8 ];
```
Comment (pbsds) [inline pkgs/development/python-modules/markdownify/default.nix:17]: Seems like the flake8 dependency is only needed for their lint command, which we don't use. Please remove it, as it reduces the closure size.

File: pkgs/development/python-modules/rsskey/default.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  pname = "rsskey";
+  version = "0.2.0";
+  format = "flit";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-QedLuwd0ES2LWhZ72Cjh3+ZZ7HbRyNsyLN9lNFbY5dQ=";
+  };
+
+  propagatedBuildInputs = [ feedparser httpx loca markdownify trio ];
```
Comment (pbsds) [inline pkgs/development/python-modules/rsskey/default.nix:19]: I'd split this over multiple lines

**Review by pbsds — CHANGES_REQUESTED** [review]
Looking good! I should've caught these last time!

File: pkgs/development/python-modules/rsskey/default.nix (hunk 4 above)
Comment (McSinyx) [reply pkgs/development/python-modules/rsskey/default.nix]: Done.

File: pkgs/development/python-modules/markdownify/default.nix (hunk 3 above)
Comment (McSinyx) [reply pkgs/development/python-modules/markdownify/default.nix]: I filed https://github.com/matthewwithanm/python-markdownify/issues/70 and added a downstream patch in the mean time.

**Review by pbsds — APPROVED** [review]

File: pkgs/development/python-modules/markdownify/default.nix (hunk 3 above)
Comment (McSinyx) [reply pkgs/development/python-modules/markdownify/default.nix]: Upstream removed the build requirement.

File: pkgs/development/python-modules/rsskey/default.nix [hunk 5]
```diff
@@ -0,0 +1,36 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, feedparser
+, httpx
+, loca
+, markdownify
+, trio }:
+buildPythonPackage rec {
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/rsskey/default.nix:9]:
```suggestion
, trio
}:

buildPythonPackage rec {
```

File: pkgs/development/python-modules/markdownify/default.nix [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+  propagatedBuildInputs = [ beautifulsoup4 six ];
+  checkInputs = [ pytestCheckHook ];
+
+  meta = with lib; {
+    description = "HTML to Markdown converter";
+    homepage = "https://github.com/matthewwithanm/python-markdownify";
+    license = licenses.mit;
+    maintainers = [ maintainers.McSinyx ];
+    platforms = platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/markdownify/default.nix:24]:
```suggestion
```

**Comment by SuperSandro2000** [comment]
@ofborg build python310Packages.rsskey

**Pull request merged**